fn/map_test.go

```diff
@@ -0,0 +1,137 @@
+package fn
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestKeySet(t *testing.T) {
+	testMap := map[string]int{"a": 1, "b": 2, "c": 3}
+	expected := NewSet([]string{"a", "b", "c"}...)
+
+	result := KeySet(testMap)
+
+	require.Equal(t, result, expected)
+}
+
+// TestNewSubMap tests the NewSubMap function with various input cases.
+func TestNewSubMap(t *testing.T) {
+	tests := []struct {
+		name     string
+		original map[int]string
+		keys     []int
+		expected map[int]string
+		wantErr  bool
+	}{
+		{
+			name: "Successful submap creation",
+			original: map[int]string{
+				1: "apple",
+				2: "banana",
+				3: "cherry",
+			},
+			keys: []int{1, 3},
+			expected: map[int]string{
+				1: "apple",
+				3: "cherry",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Key not found",
+			original: map[int]string{
+				1: "apple",
+				2: "banana",
+			},
+			keys:     []int{1, 4},
+			expected: nil,
+			wantErr:  true,
+		},
+		{
+			name: "Empty keys list",
+			original: map[int]string{
+				1: "apple",
+				2: "banana",
+			},
+			keys:     []int{},
+			expected: map[int]string{},
+			wantErr:  false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := NewSubMap(tt.original, tt.keys)
+			if tt.wantErr {
+				require.Contains(
```

Comment:
I think this should be `require.ErrContains()` instead, otherwise this will panic if we want an err but it is `nil`.